lib/chess/functional/cell_notation.rb

```diff
@@ -0,0 +1,29 @@
+require_relative '../core/chess'
+
+module Chess
+  module Functional
+    module CellNotation
+      Algebraic = Struct.new(:column, :row) do
+        def to_s
+          "#{column}#{row}"
+        end
+
+        def to_cartesian_a
+          [row - 1, COLUMNS.index(column)]
+        end
```

Comment:
This seems dirty. The Algebraic class shouldn't know how a Cartesian object can be parsed to an array.

I think the Algebraic class should have a `to_cartesian` method that converts the algebraic notation to cartesian, and if you get to need to convert an algebraic object to cartesian to array, you just do `algebraic.to_cartesian.to_a`